guardrails/validator_service.py

```diff
@@ -142,7 +143,7 @@ def after_run_validator(
         self,
         validator: Validator,
         validator_logs: ValidatorLogs,
-        result: ValidationResult,
+        result: ValidationResult | None,
```

Comment:
Use `Optional[ValidationResult]` instead.  `|` unions aren't supported in lower python versions.